src/main/kotlin/com/aemtools/completion/util/HtlUtil.kt

```diff
@@ -31,4 +34,47 @@ fun extractItemAndItemListNames(value: String): Pair<String, String> {
         }
     }
     return item to "${item}List"
+}
+
+/**
+ * Check if current string literal is the main EL's string
+ * (e.g. ${'main string' @ param='not main string'})
+ */
+fun HtlStringLiteral.isMainString(): Boolean {
+    val expression = this.findParentByType(com.aemtools.lang.htl.psi.HtlExpression::class.java) ?: return false
+
+    return expression.parent is HtlHel
+}
+
+/**
+ * Check if current variable is "option"
+ */
+fun HtlVariableName.isOption() : Boolean {
+    return this.hasParent(HtlContextExpression::class.java)
+        && !this.hasParent(HtlAssignment::class.java)
+}
+
+/**
+ * Extract first (top level) [PropertyAccessMixin].
+ */
+fun HtlHtlEl.extractPropertyAccess(): PropertyAccessMixin? {
```

Comment:
What if you will use PciElement oe just Element, will PCI generator add Htl prefix? So it will be HtlPciElement or HtlElement. Will it work for you?